### PR TITLE
Rework info.json reference

### PR DESCRIPTION
**Pull request by fauxpark**
## Description

I think I got pretty much everything.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by dunk2k** [comment]
@fauxpark Great write up chief!

Could I be a pain with a question:
    Should `development_board` be used to indicate that keyboard requires a microcontroller (daughter) board exclusively?. i.e. the controller chip **is not** part of the key switch/mother PCB.

File: docs/reference_info_json.md [hunk 1]
```diff
@@ -1,369 +1,693 @@
...
+
+## Hardware Configuration :id=hardware-configuration
+
+* `board`
+    * Override the default ChibiOS board name (ARM-based keyboards only).
+    * Example: `"BLACKPILL_STM32_F411"`
+* `bootloader`
+    * The bootloader in use on the keyboard. Required if `development_board` is not specified.
+* `development_board`
+    * The the microcontroller development board, if applicable.
```
Comment (dunk2k) [inline docs/reference_info_json.md:36]:
```suggestion
    * The microcontroller development board, if applicable.
```

Comment (sigprof) [inline docs/reference_info_json.md:36]:
```suggestion
    * The microcontroller development board, if applicable.
```

**Review by sigprof — CHANGES_REQUESTED** [review]

**Comment by fauxpark** [comment]
@dunk2k Are you asking whether that should be documented here, or are you asking when to use `development_board`?

Since this page is only meant as a quick reference, with each section linking to the respective feature/driver/etc. docs, it's not really in scope to provide more than a simple description of a given key. I've also omitted the full list of MCUs and bootloaders for a similar reason.

**Comment by dunk2k** [comment]
> @dunk2k Are you asking whether that should be documented here, or are you asking when to use `development_board`?

The latter. Though if there's documentation in pipeline/future that describes when to use `development_board` then that document will save someone asking to elaborate in this file 😁

**Comment by fauxpark** [comment]
Then to answer your original question, yes. In this case you would not need to specify `processor` and `bootloader` (and usually also `board` for ChibiOS targets): https://github.com/qmk/qmk_firmware/blob/develop/data/mappings/defaults.hjson

**Review by drashna — APPROVED** [review]

**Comment by dunk2k** [comment]
Would a listing of all available entries for `"features"`, be appropriate in this file?

**Comment by fauxpark** [comment]
There is no canonical list. Most of them, I think, should be moved into their own `<feature>.enabled` key anyway.

**Review by dunk2k — APPROVED** [review]

File: docs/reference_info_json.md [hunk 2]
```diff
@@ -1,369 +1,696 @@
...
+            * See [Bootmagic](#bootmagic) config.
+    * `enabled`
+        * Enable the Split Keyboard feature.
+        * Default: `false`
+    * `encoder`
+        * `right`
+            * `rotary`
+                * See [Encoder](#encoder) config.
+    * `main`
+        * How to determine the main/primary half. Must be one of `eeprom`, `left`, `matrix_grid`, `pin`, `right`.
```
Comment (zvecr) [inline docs/reference_info_json.md:595]: Probably not worth documenting this given its broken, <https://github.com/qmk/qmk_firmware/pull/18254>. Same goes for some of the other fields that PR flags issue to, matrix_grid.

**Review by zvecr — APPROVED** [review]

**Pull request merged**